nimbess-cni.go

```diff
@@ -0,0 +1,314 @@
+		})
+		for _, ip := range iface.IpAddresses {
+			// append interface ip address info
+			_, ipAddr, err := net.ParseCIDR(ip.Address)
+			if err != nil {
+				log.Error(err)
+				return err
+			}
+			var gwAddr net.IP
+			if ip.Gateway != "" {
+				gwAddr = net.ParseIP(ip.Gateway)
+				if err != nil {
+					log.Error(err)
+					return err
+				}
+			}
+			ver := "4"
+			if ip.Version == cninimbess.CNIReply_Interface_IP_IPV6 {
+				ver = "6"
+			}
+			cniResult.IPs = append(cniResult.IPs, &cnitypes.IPConfig{
+				Address:   *ipAddr,
+				Version:   ver,
+				Interface: &ifidx,
+				Gateway:   gwAddr,
+			})
+		}
+	}
+
+	// Process routes
+	for _, route := range r.Routes {
+		_, dstIP, err := net.ParseCIDR(route.Dst)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+		gwAddr := net.ParseIP(route.Gw)
+		if err != nil {
+			log.Error(err)
+			return err
+		}
+		cniResult.Routes = append(cniResult.Routes, &types.Route{
+			Dst: *dstIP,
+			GW:  gwAddr,
+		})
+	}
+
+	// Process DNS entry
+	for _, dns := range r.Dns {
+		cniResult.DNS.Nameservers = dns.Nameservers
+		cniResult.DNS.Domain = dns.Domain
+		cniResult.DNS.Search = dns.Search
+		cniResult.DNS.Options = dns.Options
+	}
+
+	log.WithFields(log.Fields{"Result": cniResult}).Debugf("CNI ADD request OK, took %s", time.Since(start))
+
+	return cniResult.Print()
+}
+
+// cmdDel implements the CNI request to delete a container from network.
+// It forwards the request to he remote gRPC server and returns the result received from gRPC.
+func cmdDel(args *skel.CmdArgs) error {
+	start := time.Now()
+
+	// Parse CNI config
+	conf, err := parseCNIConfig(args.StdinData)
+	if err != nil {
+		log.Errorf("Unable to parse CNI config: %v", err)
+		return err
+	}
+
+	err = initLog(conf.LogFile)
+	if err != nil {
+		log.Errorf("Unable to initialize logging: %v", err)
+		return err
+	}
+	log.WithFields(log.Fields{
+		"ContainerID": args.ContainerID,
+		"Netns":       args.Netns,
+		"IfName":      args.IfName,
+		"Args":        args.Args,
+	}).Debug("CNI DEL request")
+
+	// connect to remote CNI handler over gRPC
+	conn, c, err := grpcConnect(conf.GrpcServer)
+	if err != nil {
+		log.Errorf("Unable to connect to GRPC server %s: %v", conf.GrpcServer, err)
+		return err
+	}
+	defer conn.Close()
+
+	// Prepare CNI Request for Network Config
+	cniRequestNW := &cninimbess.CNIRequest_NetworkConfig{}
+
+	// execute the DELETE request
+	_, err = c.Delete(context.Background(), &cninimbess.CNIRequest{
+		Version:          conf.CNIVersion,
+		ContainerId:      args.ContainerID,
+		InterfaceName:    args.IfName,
+		NetworkNamespace: args.Netns,
+		NetworkConfig:    cniRequestNW,
+		ExtraArguments:   cniRequestNW.Args,
+	})
+	if err != nil {
+		log.Errorf("Error by executing remote CNI Delete request: %v", err)
+		return err
+	}
+
+	// TODO: handle IPAM deletion of IPs
+
+	log.Debugf("CNI DEL request OK, took %s", time.Since(start))
+
+	return nil
+}
+
+//cmdCheck implements the CNI request to check the
```

Comment:
Is this a full comment? Or just haven't filled in the implementation yet?

Reply:
Missing the end of the comment. It needs to be implemented